apps/src/lab2/projects/utils.ts

```diff
@@ -75,6 +76,18 @@ export function getAppOptionsViewingExemplar(): boolean | undefined {
     return appOptions.isViewingExemplar;
   }
 }
+
+/**
+ * Returns the value of userSharingDisabled provided by App Options, if available.
+ * This can be used to tell if the current user has sharing disabled for advanced projects.
+ */
+export function getAppOptionsUserSharingDisabled(): boolean | undefined {
```

Comment:
Where will this be used?

Reply:
Does this need to be in app options? Is it something we could put in `currentUserRedux` instead, for example?

Reply:
I considered that, but the value is already available for legacy lab projects in `appOptions`. Also there seemed to be a hesitation to use `currentUserRedux` in the past when we were deciding on where to put font size user preferences.
But I'm open to it! I don't have strong feelings either way.

Reply:
I think if we don't need to use app options, we shouldn't. It doesn't necessarily have to go in current user redux, although I think that may make the most sense?

Reply:
Sure. I'm good with that and I'll update. But do you mind explaining when we should use app options and when we shouldn't? Thanks!

Reply:
Basically if we absolutely have to use app options for compatibility with old labs we should, but otherwise we try to be more modern and use api requests to get user/level data. Does that sound like the right summary @sanchitmalhotra126?